src/backend/aspen/workflows/pangolin/save.py

```diff
@@ -15,14 +15,19 @@
 from aspen.database.models import PathogenGenome, UploadedPathogenGenome
 
 
-def get_probability(ambiguity_score: float) -> int:
+def get_probability(row: dict) -> Union[int, None]:
     """
     Estimate confidence percentage based on the `ambiguity_score`.
     Per pangolin docs, this is basically the number of lineage-defining sites
     that had to be imputed from the reference sequence.
     Round and multiply by 100 --> percentage for easier user comprehension.
     """
-    return round(ambiguity_score * 100)
+    if row["ambiguity_score"] != "":
+        return round(row["ambiguity_score"] * 100)
```

Comment:
I think you still need the float cast here? `round(float(row["ambiguity_score"]) * 100)`

Reply:
Maybe? Only ever encountered values that are floats or strings here, but agree it's safer to explicitly call it out. Will fix!